packages/common/src/index.ts

```diff
@@ -1124,29 +1124,32 @@ function formatRawValue(
         // We want to return the datetime in UTC to avoid timezone issues in the frontend like in chart tooltips
         return dayjs(value).utc(true).format();
     }
+
     return value;
 }
 
-export function formatRows(
-    rows: { [col: string]: any }[],
-    itemsMap: ItemsMap,
-): ResultRow[] {
-    return rows.map((row) =>
-        Object.keys(row).reduce<ResultRow>((acc, columnName) => {
-            const col = row[columnName];
-
-            const item = itemsMap[columnName];
-            return {
-                ...acc,
-                [columnName]: {
-                    value: {
-                        raw: formatRawValue(item, col),
-                        formatted: formatItemValue(item, col),
-                    },
+export function formatRows(rows: { [col: string]: any }[], itemsMap: ItemsMap) {
+    const columnNames = Object.keys(rows[0] || {});
```

Comment:
is it possible that there are more column names on row 2?

Reply:
Done here: [chore: fetch column names per row](https://github.com/lightdash/lightdash/pull/12983/commits/d712fdfa527b0f7180ce1d2e495e4621628ee5d2)

![image](https://github.com/user-attachments/assets/88217939-48e3-4bd5-b414-cf8398ccf1e7)